datafusion/datasource-parquet/src/row_filter.rs

```diff
@@ -537,12 +464,20 @@ pub fn build_row_filter(
     // `a = 1 AND b = 2 AND c = 3` -> [`a = 1`, `b = 2`, `c = 3`]
     let predicates = split_conjunction(expr);
 
+    let file_schema = Arc::new(file_schema.clone());
+    let table_schema = Arc::new(table_schema.clone());
```

Comment:
It would have been nice to just change the caller to pass in an `Arc<>` but that would be a breaking change, this seems cheap enough to not worry too much about it.